libs/base/Data/SnocList.idr

```diff
@@ -15,13 +15,13 @@ sx <>< (x :: xs) = sx :< x <>< xs
 ||| 'chips': Action of snoc-lists on lists
 public export
 (<>>) : SnocList a -> List a -> List a
-Empty     <>> xs = xs
+Lin       <>> xs = xs
 (sx :< x) <>> xs = sx <>> x :: xs
 
 ||| Reverse the snoc-list and make a list.
 public export
 toList : SnocList type -> List type
```

Comment:
I don't think we should have local functions called `toList` because that clashes with the
`Foldable` definition. I have had issues with this in the past in `Data.List1` (hence the
function from `List1 a` to `List a` being called `forget`).

Reply:
I'm happy to drop it, it was in @jfdm 's library, but since we can recover it as `toList sx = sx <>> []`, we can just drop it.

Reply:
`sx <>> []` reverses the list whereas  ̀toList` preserves the ordering.

Reply:
Oh, do you think we should swap `cast` and `asList`?

My principle was that `cast` should maintain the logical structure.